### PR TITLE
executor: add more statements for metric of statement counter

### PR DESCRIPTION
**Pull request by XuHuaiyu**
PTAL @coocood @shenli @hanfei1991 @tiancaiamao @zimulala

### CODE REVIEW

File: executor/compiler.go [hunk 1]
```diff
@@ -30,32 +30,79 @@ type Compiler struct {
...
 		return "DropDatabase"
-	case *ast.DropTableStmt:
-		return "DropTable"
 	case *ast.DropIndexStmt:
 		return "DropIndex"
+	case *ast.DropTableStmt:
+		return "DropTable"
+	case *ast.DropUserStmt:
+		return "DropUser"
+	case *ast.ExecuteStmt:
```
Comment (shenli) [inline executor/compiler.go]: For ExecuteStmt, can you distinguish between different statements?

**Comment by shenli** [comment]
@coocood PTAL

File: executor/compiler.go [hunk 2]
```diff
@@ -30,32 +30,79 @@ type Compiler struct {
...
+	case *ast.SetPwdStmt:
+		return "SetPwd"
+	case *ast.ShowStmt:
+		return "Show"
+	case *ast.TruncateTableStmt:
+		return "TruncateTable"
+	case *ast.UpdateStmt:
+		return "Update"
 	}
 	return "unknown"
```
Comment (coocood) [inline executor/compiler.go]: rename "unknown" to "other" is better.

File: executor/compiler.go [hunk 3]
```diff
@@ -30,32 +30,79 @@ type Compiler struct {
...
+		return "LoadData"
+	case *ast.PrepareStmt:
+		return "Prepare"
+	case *ast.RollbackStmt:
+		return "Rollback"
+	case *ast.SelectStmt:
+		if x.From == nil {
+			return "Select-Simple"
+		}
+		fileds := x.From.TableRefs
```
Comment (coocood) [inline executor/compiler.go]: Why name it "fileds"?

File: executor/compiler.go [hunk 4]
```diff
@@ -30,32 +30,79 @@ type Compiler struct {
...
+			switch ref := fileds.Left.(type) {
+			case *ast.TableSource:
+				switch ref.Source.(type) {
+				case *ast.TableName:
+					return "Select-Simple"
+				}
+			}
+		}
+		return "Select-Complex"
+	case *ast.SetPwdStmt:
```
Comment (coocood) [inline executor/compiler.go]: `SetPwdStmt` is not needed.

**Comment by XuHuaiyu** [comment]
@shenli @coocood PTAL

**Comment by coocood** [comment]
`DeallocateStmt` is the opposite of `PrepareStmt`, should be ignored too.

`DoStmt`,  `GrantStmt`, `FlushTableStmt`, `DropUserStmt` are rarely used,  can be leaved to `other`.

**Comment by coocood** [comment]
LGTM

File: executor/adapter.go [hunk 5]
```diff
@@ -100,6 +100,7 @@ func (a *statement) Exec(ctx context.Context) (ast.RecordSet, error) {
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
+		stmtCount(executorExec.Stmt)
```
Comment (shenli) [inline executor/adapter.go]: Could you add a test case for this?

File: executor/compiler.go [hunk 6]
```diff
@@ -30,41 +30,76 @@ type Compiler struct {
...
+	case *ast.SetStmt, *ast.SetPwdStmt:
+		return "Set"
+	case *ast.ShowStmt:
+		return "Show"
+	case *ast.TruncateTableStmt:
+		return "TruncateTable"
+	case *ast.UpdateStmt:
+		return "Update"
+	case *ast.DeallocateStmt, *ast.ExecuteStmt, *ast.PrepareStmt, *ast.UseStmt:
+		return "Ignored"
```
Comment (shenli) [inline executor/compiler.go]: Use a const for "Ignored". Because this is used in other place.

File: executor/compiler.go [hunk 7]
```diff
@@ -30,41 +30,76 @@ type Compiler struct {
...
+		if tableRefs.Right == nil {
+			switch ref := tableRefs.Left.(type) {
+			case *ast.TableSource:
+				switch ref.Source.(type) {
+				case *ast.TableName:
+					return "Select-Simple"
+				}
+			}
+		}
+		return "Select-Complex"
```
Comment (zimulala) [inline executor/compiler.go]: Can this be extracted into a function?

File: executor/compiler.go [hunk 8]
```diff
@@ -30,41 +30,76 @@ type Compiler struct {
...
-	return "unknown"
+	return "other"
 }
 
 // Compile compiles an ast.StmtNode to an ast.Statement.
 // After preprocessed and validated, it will be optimized to a plan,
 // then wrappped to an adapter *statement as stmt.Statement.
 func (c *Compiler) Compile(ctx context.Context, node ast.StmtNode) (ast.Statement, error) {
-	stmtNodeCounter.WithLabelValues(statementLabel(node)).Inc()
+	stmtCount(node)
```
Comment (tiancaiamao) [inline executor/compiler.go]: `stmtCount` is called in both `statement.Exec`  and here, would the metrics be calculated twice?

**Comment by XuHuaiyu** [comment]
@zimulala @shenli @coocood PTAL

File: executor/compiler.go [hunk 9]
```diff
@@ -67,20 +72,7 @@ func statementLabel(node ast.StmtNode) string {
...
-			switch ref := tableRefs.Left.(type) {
-			case *ast.TableSource:
-				switch ref.Source.(type) {
-				case *ast.TableName:
-					return "Select-Simple"
-				}
-			}
-		}
-		return "Select-Complex"
+		return distinguishSelectStmt(x)
```
Comment (ngaut) [inline executor/compiler.go]: s/distinguishSelectStmt/selectStmt2Label/

Comment (ngaut) [inline executor/compiler.go]: or getSelectstmtLabel

File: executor/compiler.go [hunk 10]
```diff
@@ -90,11 +82,28 @@ func statementLabel(node ast.StmtNode) string {
...
+	if x.From == nil {
+		return "Select-Simple"
+	}
+	tableRefs := x.From.TableRefs
+	if tableRefs.Right == nil {
+		switch ref := tableRefs.Left.(type) {
+		case *ast.TableSource:
+			switch ref.Source.(type) {
+			case *ast.TableName:
+				return "Select-Simple"
```
Comment (ngaut) [inline executor/compiler.go]: Also define as constant string.

**Comment by XuHuaiyu** [comment]
@shenli @ngaut PTAL

File: server/server_test.go [hunk 11]
```diff
@@ -517,3 +521,54 @@ func runTestMultiStatements(c *C) {
...
+	resp, err := http.Get("http://127.0.0.1:10090/metrics")
+	t.Assert(err, IsNil)
+	content, err := ioutil.ReadAll(resp.Body)
+	t.Assert(err, IsNil)
+	resp.Body.Close()
+	return content
+}
+
+func getStmtCnt(content string) (stmtCnt map[string]int) {
+	stmtCnt = make(map[string]int, 0)
```
Comment (ngaut) [inline server/server_test.go]: Just simply make map without capacity.

**Comment by ngaut** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**